vm_supervisor/ipfs.py

```diff
@@ -0,0 +1,66 @@
+import logging
+from pathlib import Path
+from typing import Optional
+
+import aioipfs
+from aleph.sdk.chains.common import get_fallback_private_key
+from aleph.sdk.chains.ethereum import ETHAccount
+from aleph.sdk.client import AuthenticatedAlephClient
+from aleph.sdk.types import StorageEnum
+from aleph_message.models import ItemHash
+from aleph_message.status import MessageStatus
+
+logger = logging.getLogger(__name__)
+
+
+async def ipfs_upload_file(path: Path) -> str:
+    """Push a file to the IPFS service."""
+    # Use local ipfs node by default. TODO: Add a general setting to allow to customize it
+    client = aioipfs.AsyncIPFS(host="localhost", port=5001)
+    ipfs_hash = None
+    async for added_file in client.add(str(path), recursive=True):
```

Comment:
Why is this recursive if the goal is to upload a single file ?

Reply:
Because it accepts different files, and it return different results.

Reply:
In practice I only see it used in one place, `snapshot_hash = await ipfs_upload_file(self.path)`, with a single file.

Reply:
Yes, I mean that the `client.add` method call, it accepts different files, and return different results.